tests/integration/security/citadel/namespace_targeting_test.go

```diff
@@ -0,0 +1,93 @@
+package citadel
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+
+	v1 "k8s.io/api/core/v1"
+	mv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	caOverrideLabel    = "ca.istio.io/override"
+	caEnvLabel         = "ca.istio.io/env"
+	testNs             = "targeting-test"
+	serviceAccountName = "sample-service-account"
+)
+
+// TestNamespaceTargeting verifies that Citadel respects the designated namespace targeting labels.
+// current implementation is closely tied with Kubernetes concepts, but this is difficult to avoid
+// until Citadel itself is less closely tied with Kubernetes concepts
+func TestNamespaceTargeting(t *testing.T) {
+	framework.NewTest(t).Run(func(ctx framework.TestContext) {
+
+		env := ctx.Environment().(*kube.Environment)
+
+		citadelNamespace := ist.Settings().SystemNamespace
+		type namespaceLabels map[string]string
+		testCases := map[string]struct {
+			labels   namespaceLabels
+			targeted bool
+		}{
+			"override label set to false, env label matches => should not generate secret": {
```

Comment:
you run test with t.Run(key_in_the_map, xx), won't that make the displayed test case hard to display?

`TestNamespaceTargeting/override label set to false, env label matches => should not generate secret`

I see other test code using short and no spaced test case name for this kinda purpose

Reply:
Fixed, made a bit more concise